tools/serve/serve.py

```diff
@@ -345,6 +345,44 @@ class ServiceWorkerModulesHandler(HtmlWrapperHandler):
 </script>
 """
 
+class ShadowRealmHandler(HtmlWrapperHandler):
+    global_type = "shadowrealm"
+    path_replace = [(".any.shadowrealm.html", ".any.js")]
+
+    wrapper = """<!doctype html>
+<meta charset=utf-8>
+%(meta)s
+<script src="/resources/testharness.js"></script>
+<script src="/resources/testharnessreport.js"></script>
+<script>
+(async function() {
+  const r = new ShadowRealm();
+
+  await new Promise(r.evaluate(`
+    (resolve, reject) => {
+      (async () => {
+        await import("/resources/testharness.js");
+        %(script)s
+        globalThis.self.GLOBAL = {
```

Comment:
`self` is not formally defined here, is it?

Reply:
It is not, but I kept this after the [previous discussion](https://github.com/web-platform-tests/wpt/pull/32956#discussion_r814228330) on the earlier version of the patch